Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreFloatMapper.java

```diff
@@ -50,7 +53,27 @@ public PointStore toModel(PointStoreState state, long seed) {
         int[] refCount = ArrayPacking.unpackInts(state.getRefCount(), indexCapacity, state.isCompressed());
         int[] locationList = new int[indexCapacity];
         int[] tempList = ArrayPacking.unpackInts(state.getLocationList(), state.isCompressed());
-        System.arraycopy(tempList, 0, locationList, 0, tempList.length);
+        if (!state.getVersion().equals(Version.V3_0)) {
```

Comment:
If we start having to create more branching logic like this, I suggest we start creating separate methods for clarity:
```
public PointStore toModel(PointStoreState state, long seed) {
  if (state.getVersion.compareTo(Version.V3_0) < 0) {
    return toModelV2(state, seed);
  } else {
    return toModelV3(state, seed);
  }
}

private PointStore toModelV2(PointStoreState state, long seed) { ... }
private PointStore toModelV3(PointStoreState state, long seed) { ... }
```
}
```

Reply:
Agree wholeheartedly. Lets even start now - while we remember changes. Will push changes in the next few PRs (maybe keep the next one focused on just delete).